src/sdk/conference/channel.js

```diff
@@ -129,14 +131,28 @@ export class ConferencePeerConnectionChannel extends EventDispatcher {
           mid: t.mid,
         };
       });
-      const publicationId =
-          await this._signaling.sendSignalingMessage('publish', {
-            media: {tracks: trackOptions},
-            attributes: stream.attributes,
-            transport: {id: this._id, type: 'webrtc'},
-          }).id;
+      const publicationResp =
+        await this._signaling.sendSignalingMessage('publish', {
+          media: {tracks: trackOptions},
+          attributes: stream.attributes,
+          transport: {id: this._id, type: 'webrtc'},
+        });
+      const publicationId = publicationResp.id;
+      const messageEvent = new MessageEvent('id', {
+        message: publicationId,
+        origin: this._remoteId,
+      });
+      this.dispatchEvent(messageEvent);
+
       this._publishTransceivers.get(internalId).id = publicationId;
       this._reverseIdMap.set(publicationId, internalId);
+
+      if (this._id && this._id !== publicationResp.transportId) {
+        Logger.warning('Server returns conflict ID: ' + publicationResp
```

Comment:
no handling here?

Reply:
I'm not sure if we should drop the response or continue the workflow. Since the client SDK uses a single PeerConnection by default, the response is likely belonging to this channel. But we can drop it if you prefer to explicitly avoid this error.

Reply:
prefer you drop it.

Reply:
Added `return` and notify `ConferenceClient` only after checking transport ID.